src/signal.cr

```diff
@@ -132,6 +137,28 @@ enum Signal : Int32
     Signal::PIPE.ignore
     Signal::CHLD.reset
   end
+
+  # :nodoc:
+  def self.setup_sigfault_handler
+    return unless @@setup_sigfault_handler.test_and_set
```

Comment:
Just to confirm: Previously there was no flag protection for `LibExt.setup_sigfault_handler
`, right? It's probably a good idea to add it, to avoid duplicate initialization :+1:

Reply:
yeah, there was none in the C extension. (Though by default initialization only happens once on the main thread, so it's probably not a huge risk, but why not do it? 😄 )

Reply:
Totally. Some user code could accidentally call `Signal.setup_sigfault_handle`.